mountpoint-s3-client/src/mock_client.rs

```diff
@@ -252,6 +269,12 @@ impl ObjectClient for MockClient {
     ) -> ObjectClientResult<Self::GetObjectResult, GetObjectError, Self::ClientError> {
         trace!(bucket, key, ?range, ?if_match, "GetObject");
 
+        if let Some(etag) = if_match {
```

Comment:
Shouldn't the etag be checked after the object is found? 

I think that also means we do not need the `get_etag` method.